internal/controller/deployment_controller.go

```diff
@@ -87,7 +91,69 @@ func (r *DeploymentReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return r.Update(ctx, l, deployment)
 }
 
+func (r *DeploymentReconciler) setStatusFromClusterStatus(ctx context.Context, dc *dynamic.DynamicClient, l logr.Logger, deployment *hmc.Deployment) (bool, error) {
+	resourceId := schema.GroupVersionResource{
+		Group:    "cluster.x-k8s.io",
+		Version:  "v1beta1",
+		Resource: "clusters",
```

Comment:
I don't want to scope creep you, perhaps it's in a different issue, but Is `clusters` the only resource that this status should be referencing?  What about `Workers` and specifically `MachineDeployment`'s, the cluster could be `ready: True` when the workers aren't yet up.

Reply:
Michael said to just go with the clusters for now.